### PR TITLE
o1vm/riscv32im: rem and remu by zero

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: o1vm/tests/test_riscv_elf.rs [hunk 1]
```diff
@@ -384,3 +384,33 @@ fn test_divu_by_zero() {
...
+    let curr_dir = std::env::current_dir().unwrap();
+    let path = curr_dir.join(std::path::PathBuf::from(
+        "resources/programs/riscv32im/bin/rem_by_zero",
+    ));
+    let state = o1vm::elf_loader::parse_riscv32(&path).unwrap();
+    let mut witness = Env::<Fp>::create(PAGE_SIZE.try_into().unwrap(), state);
+
+    while !witness.halt {
+        witness.step();
+    }
```
Comment (svv232) [inline o1vm/tests/test_riscv_elf.rs:400]: missing assertions

File: o1vm/tests/test_riscv_elf.rs [hunk 2]
```diff
@@ -384,3 +384,33 @@ fn test_divu_by_zero() {
...
+    let curr_dir = std::env::current_dir().unwrap();
+    let path = curr_dir.join(std::path::PathBuf::from(
+        "resources/programs/riscv32im/bin/remu_by_zero",
+    ));
+    let state = o1vm::elf_loader::parse_riscv32(&path).unwrap();
+    let mut witness = Env::<Fp>::create(PAGE_SIZE.try_into().unwrap(), state);
+
+    while !witness.halt {
+        witness.step();
+    }
```
Comment (svv232) [inline o1vm/tests/test_riscv_elf.rs:415]: missing assertions

**Review by svv232 — CHANGES_REQUESTED** [review]

File: o1vm/tests/test_riscv_elf.rs (hunk 1 above)
Comment (dannywillems) [reply o1vm/tests/test_riscv_elf.rs:400]: No, the idea is that it should fail. See `#[should_panic]`

File: o1vm/tests/test_riscv_elf.rs (hunk 2 above)
Comment (dannywillems) [reply o1vm/tests/test_riscv_elf.rs:415]: No, the idea is that it should fail. See `#[should_panic]`

**Review by svv232 — APPROVED** [review]

**Pull request merged**